connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -2597,6 +2600,50 @@ class SparkConnectPlanner(val session: SparkSession) {
         .build())
   }
 
+  /**
+   * A helper function to handle streaming awaitTermination(). awaitTermination() can be a long
+   * running command. In this function, we periodically check if the RPC call has been cancelled.
+   * If so, we can stop the operation and release resources early.
+   * @param query the query waits to be terminated
+   * @param timeoutMs optional. Timeout to wait for termination. If None, no timeout is set
+   * @return if the query has terminated
+   */
+  private def handleStreamingAwaitTermination(
+      query: StreamingQuery,
+      timeoutMs: Option[Long]): Boolean = {
+    // How often to check if RPC is cancelled and call awaitTermination()
+    val awaitTerminationIntervalMs = 10000
+
+    val hasTimeout = timeoutMs.isDefined
+    var timeoutLeftMs = timeoutMs.getOrElse(Long.MaxValue)
+    require(timeoutLeftMs > 0, "Timeout has to be positive")
+
+    val grpcContext = Context.current
+    while (!grpcContext.isCancelled) {
+      val awaitTimeMs = if (hasTimeout) {
```

Comment:
No need to check, timeout is already set to max value when it is not set.

Reply:
done